common/src/point_types.cpp

```diff
@@ -70,6 +70,13 @@ namespace pcl
     return (os);
   }
 
+  std::ostream&
+  operator << (std::ostream& os, const Intensity32u& p)
+  {
+    os << "( " << static_cast<int>(p.intensity) << " )";
```

Comment:
We don't want this cast. In the case of `Intensity8u` it's needed because otherwise the value will be output as a character.

Reply:
@taketwo 
thank you for the review.
Then why is it converted from float to int for the code below
https://github.com/PointCloudLibrary/pcl/blob/8e74b8a4bb03586e285f187303f033e9d983b175/common/src/point_types.cpp#L59-L64

Reply:
This I can not say, either a copy/paste or the author wanted to omit fractional part. But for the line in question, casting to signed integer does not make sense. If the stored intensity value is a very large number, say `0xFFFFFFFF`, then an overflow will happen and we will get `-1` printed.

Reply:
@taketwo 
Fixed!